pkg/sharding/sharding.go

```diff
@@ -6,11 +6,18 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 )
 
-const ShardingLabel string = "fleet.cattle.io/shard"
+const (
+	// ShardingIDLabel is the label key used to identify the shard ID of a controller pod
+	ShardingIDLabel string = "fleet.cattle.io/shard-id"
+	// ShardingRefLabel is the label key used by resources to reference the shard ID of a controller
+	ShardingRefLabel string = "fleet.cattle.io/shard-ref"
+	// ShardingDefaultLabel is the label key which is set to true on the controller handling unlabeled resources
+	ShardingDefaultLabel string = "fleet.cattle.io/shard-default"
+)
 
 func FilterByShardID(shardID string) predicate.Funcs {
 	matchesLabel := func(o client.Object) bool {
-		label, hasLabel := o.GetLabels()[ShardingLabel]
+		label, hasLabel := o.GetLabels()[ShardingIDLabel]
```

Comment:
Since this check happens on deployed resources, I think we'd want this instead:
```suggestion
		label, hasLabel := o.GetLabels()[ShardingRefLabel]
```